### PR TITLE
Add gwsurrogate

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This commit adds recipes for `gwtools` and `gwsurrogate`.

@sfield17, @vijayvarma392, can you please post messages stating that you are ok being named as co-maintainers of these recipes? thanks.

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
@sfield17, can you suggest a `home` URL for the `gwsurrogate` package?

**Comment by vijayvarma392** [comment]
I am ok being named as a co-maintainer of these recipes.

**Comment by duncanmmacleod** [comment]
@conda-forge/help-python, the [linux build](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=76581&view=logs&jobId=10bda42a-d420-5c28-a43f-bc4e66405e5c) is failing with a package conflict, reminiscent of what @beckermr helped diagnose from #9546, is this the same thing, or a real failure?

File: recipes/gwsurrogate/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  skip: true  # [win]
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - gsl
+    - gwtools
+    - numpy
```
Comment (scopatz) [inline recipes/gwsurrogate/meta.yaml]: If you build against numpy, you should follow these instructions: https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=numpy#building-against-numpy

Comment (duncanmmacleod) [reply recipes/gwsurrogate/meta.yaml]: @scopatz, I have a `pin_compatible()` in the right place in the `run` list (AFAICT). What else is missing? Sorry if I'm missing something obvious.

**Comment by scopatz** [comment]
Oh hmmm maybe one of the deps here needs a new build.

**Comment by sfield17** [comment]
I am ok being named as a co-maintainer of these recipes too.

@duncanmmacleod Is it possible to list the home as https://pypi.org/project/gwsurrogate/ or is home more commonly at code hosting URL like github?

**Comment by duncanmmacleod** [comment]
So, building `gwsurrogate` on its own didn't solve the problem like I thought it would. Must be an issue with the dependencies. I can't reproduce this outside of the `staged-recipes` docker build, so I presume is something to do with `staged-recipes`' build config.

**Comment by duncanmmacleod** [comment]
@conda-forge/staged-recipes, I have had to manually pin the `host` requirements to `numpy 1.14` because of this line in the `build_steps.sh`:

https://github.com/conda-forge/staged-recipes/blob/0c2ee264eb6d945da846bc64aafb60177baa0ae1/.circleci/build_steps.sh#L40

Since `conda-forge-pinning` now uses `numpy 1.14` I presumably shouldn't need that in the actual feedstock. Is it feasible to either modify or remove the above variable, or patch the staged-recipes builder to use `conda-forge-pinning` to make it more feedstocky?

**Comment by duncanmmacleod** [comment]
@conda-forge/help-python, this recipe is ready for review, thanks.

**Comment by duncanmmacleod** [comment]
> Can you try removing the environment variable and your pin to check which version is picked up?

1.14.6

**Comment by duncanmmacleod** [comment]
@conda-forge/help-python, @conda-forge/staged-recipes, this recipe is ready for review now, thanks.

**Pull request merged**